### PR TITLE
Use git remote url to determine nwo

### LINKED ISSUE #44: Try to get repository name from git branch, when available
Rather than requiring `repository` in the config, try to get the repo name from git's `origin` branch, when a site's using Git.

### PR DESCRIPTION
**Pull request by benbalter**
Fixes #44

@parkr any idea how to properly mock the `git_remote_url` method?

### CODE REVIEW

File: spec/ghp_metadata_generator_spec.rb [hunk 1]
```diff
@@ -18,10 +18,17 @@
...
+    context "without a git nwo" do
+      it "raises a NoRepositoryError" do
+        expect(-> {
+          subject.send(:nwo, site)
+        }).to raise_error(Jekyll::GitHubMetadata::NoRepositoryError)
+      end
+    end
+
+    it "grabs the git nwo from an HTTPS url" do
+      subject.class.any_instance.stub(:git_remote_url) { "htps://github.com/foo/bar" }
```
Comment (parkr) [inline spec/ghp_metadata_generator_spec.rb]: `allow(subject).to receive(:git_remote_url).and_return("https://github.com/foo/bar")`

**Comment by benbalter** [comment]
Ahh. Google was failing me. Thanks @parkr fixed.

**Comment by benbalter** [comment]
To note, my motivation for this, is if this plugin is included by default via the pages gem, that means that without this fix, sites will fail to build by default, since they don't specify a repository in the config.

File: spec/ghp_metadata_generator_spec.rb [hunk 2]
```diff
@@ -18,10 +18,29 @@
...
+      it "raises a NoRepositoryError" do
+        allow(subject).to receive(:git_remote_url).and_return("")
+        expect(-> {
+          subject.send(:nwo, site)
+        }).to raise_error(Jekyll::GitHubMetadata::NoRepositoryError)
+      end
+    end
+
+    it "retrieves the git remote" do
+      expect(subject.send(:git_remote_url)).to include("jekyll/github-metadata")
```
Comment (parkr) [inline spec/ghp_metadata_generator_spec.rb]: This seems to fail. Think it's hitting your stub above?

https://travis-ci.org/jekyll/github-metadata/jobs/117149521#L232

**Comment by parkr** [comment]
@benbalter Due to https://github.com/git/git/commit/96f78d39989d1fcf393d7bc42357467dd8cf0f15#diff-388838443870b4d1c62785151773c6ff being _relatively_ new, I think going the route of the olden days (`git remote -v` and grepping) is our best bet for greatest compatibility.

**Comment by parkr** [comment]
@jekyllbot: merge

**Comment by ctubbsii** [comment]
What version is this fixed in? I don't see a milestone marked on the GitHub issue.

**Comment by parkr** [comment]
@ctubbsii https://github.com/jekyll/github-metadata/releases/tag/v1.10.0

**Comment by ctubbsii** [comment]
Looks like this is still a problem with auto-rebuilds when running `jekyll serve`, at least with jekyll-github-metadata 2.2.0.

**Pull request merged**